### PR TITLE
Refactor CSS preprocessing handling

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

Refactor CSS preprocessing to use Vite's `preprocessCSS` api so we can avoid hacking into Vite's `vite:css` plugin.

## Testing

All existing tests should pass.

## Docs

N/A. internal refactor.

### CODE REVIEW

File: packages/astro/src/core/compile/style.ts [hunk 1]
```diff
@@ -31,13 +36,65 @@ export function createStylePreprocessor(
...
+			try {
+				err = enhanceCSSError(err, filename);
+			} catch {}
+			cssTransformErrors.push(err);
+			return { error: err + '' };
 		}
 	};
+}
+
+function enhanceCSSError(err: any, filename: string) {
```
Comment (bluwy) [inline packages/astro/src/core/compile/style.ts:49]: This function is largely copied over from @Princesseuh's CSS error handling previously in `packages/astro/src/vite-style-transform/style-transform.ts`

**Comment by bluwy** [comment]
It looks like the CSS changes makes preprocessing CSS slower that it's timing out the CI? It may be indirectly fixed by #5198 increase in timeout, but I'll double-check what's causing the potential perf issue. Marking as draft.

**Comment by bluwy** [comment]
Increased the unit test timeout to 30000ms, same as #5198. Turns out it's not a perf regression, but only now that the "Invalid CSS" unit test runs the Vite style transform that increases processing time a bit.

File: packages/astro/src/core/compile/style.ts [hunk 2]
```diff
@@ -1,27 +1,32 @@
...
-
+		const id = `${filename}?astro&type=style&lang${lang}`;
 		try {
-			const result = await transformStyle(value, lang);
-
-			if (!result) return null as any; // TODO: add type in compiler to fix "any"
+			const result = await preprocessCSS(content, id, viteConfig);
 
-			for (const dep of result.deps) {
+			result.deps?.forEach((dep) => {
```
Comment (bholmesdev) [inline packages/astro/src/core/compile/style.ts:25]: nit: any reason to refactor to a `forEach` here?

**Review by bholmesdev — APPROVED** [review]
I LOVE seeing these files disappear. CSYes 👏

File: packages/astro/src/core/compile/style.ts (hunk 2 above)
Comment (bluwy) [reply packages/astro/src/core/compile/style.ts:25]: `result.deps` is now nullable so `for const` doesn't work too nice with it without wrapping an `if` condition. Thought a `?.forEach` would be cleaner 😄

**Pull request merged**